frontend/src/components/Menu/DropdownMenu.tsx

```diff
@@ -0,0 +1,102 @@
+import {
+  Box,
+  Icon,
+  Menu as ChakraMenu,
+  MenuButton as ChakraMenuButton,
+  MenuItem as ChakraMenuItem,
+  MenuItemProps,
+  MenuList as ChakraMenuList,
+  MenuListProps,
+  MenuProps,
+  useMultiStyleConfig,
+} from '@chakra-ui/react'
+
+import { BxsChevronDown } from '~/assets/icons/BxsChevronDown'
+import { BxsChevronUp } from '~/assets/icons/BxsChevronUp'
+
+export interface DropdownMenuButtonProps {
+  /**
+   * If true, the button will have a clear border
+   */
+  isClear?: boolean
+  /**
+   * Whether the menu has been opened
+   */
+  isActive: boolean
+  /**
+   * If true, the menu will take the full width of its container
+   */
+  isStretch?: boolean
+  /**
+   * Menu title
+   */
+  children: string
+}
+
+/**
+ * @preconditions Must be a child of DropdownMenu component and returned
+ * using a render prop (see implementation in DropdownMenu.stories)
+ */
+export const DropdownMenuButton = ({
+  isClear,
+  children,
+  isActive,
+  isStretch,
+}: DropdownMenuButtonProps): JSX.Element => {
+  const styles = useMultiStyleConfig('Menu', {
+    isActive,
+  })
+  return (
+    <ChakraMenuButton
+      padding={isActive ? '0.4375rem 0.9375rem' : '0.5rem 1rem'}
+      borderWidth={isActive ? '0.125rem' : '0.0625rem'}
+      borderColor={isClear ? 'white' : 'secondary.500'}
+      minWidth={isStretch ? '100%' : 'max-content'}
+      _hover={{ borderColor: isClear ? 'white' : 'secondary.700' }}
```

Comment:
These should all be in the theme, shouldn't it?

Reply:
Looking at the design and the [ChakraUI documentation](https://chakra-ui.com/docs/overlay/menu#usage), I don't actually think we need to have custom styles at all for the menu button at all. Looks like the theming could just be:

```tsx
import Button from '~components/Button'
...
// Outline variant
const DropdownMenuButton = (props) => {
  const ChevronIcon = isActive ? BxsChevronUp : BxsChevronDown
  return (
    <ChakraMenuButton 
      as={Button} 
      variant="outline" 
      colorScheme="secondary" 
      rightIcon={<ChevronIcon fontSize="1.5rem" />}
      {...props}
    />
  )
}
```